sdk/iot/Azure.Iot.Hub.Service/tests/E2eTestBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Iot.Hub.Service.Tests
+{
+    /// <summary>
+    /// This class will initialize all the settings and create and instance of the IoT Hub service client.
+    /// </summary>
+    [Parallelizable(ParallelScope.Self)]
+    public abstract class E2eTestBase : RecordedTestBase<IotHubServiceTestEnvironment>
+    {
+        public E2eTestBase(bool isAsync)
+         : base(isAsync, TestSettings.Instance.TestMode)
+        {
+        }
+
+        public E2eTestBase(bool isAsync, RecordedTestMode testMode)
+           : base(isAsync, testMode)
+        {
+        }
+
+        [SetUp]
+        public virtual void SetupE2eTestBase()
+        {
+            TestDiagnostics = false;
+
+            // TODO: set via client options and pipeline instead
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+        }
+
+        protected IoTHubServiceClient GetClient()
+        {
+            return InstrumentClient(
```

Comment:
`InstrumentClient` is giving me a NPE; I might not have set this up properly. Returning `new IoTHubServiceClient(TestSettings.Instance.IotHubConnectionString)` directly works as expected.
/cc @vinagesh , @azabbasi  for assistance.

Reply:
You also need a test settings file. And I believe some environment variables need to be setup as well. I'll have to get your branch and see what all is needed. Lets look at it together tomorrow.

Reply:
NPE? Null Pointer Exception?

Reply:
oops, null reference exception? I just love java so much! :)

Reply:
Tracked here: https://msazure.visualstudio.com/One/_workitems/edit/7112055